server/src/main/java/org/opensearch/cluster/decommission/DecommissionService.java

```diff
@@ -144,12 +145,19 @@ public void startDecommissionAction(
             public ClusterState execute(ClusterState currentState) {
                 // validates if correct awareness attributes and forced awareness attribute set to the cluster before starting action
                 validateAwarenessAttribute(decommissionAttribute, awarenessAttributes, forcedAwarenessAttributes);
+                if (decommissionRequest.requestID() == null) {
+                    decommissionRequest.setRequestID(UUIDs.randomBase64UUID());
```

Comment:
instead of setting here. Just generate it in RestAction so that you don't need it set it again. Then your request would also be immutable. Thoughts?

Reply:
In that case, only the REST layer would be having UUIDs set. We would want the IDs to be there not conditionally. When client pkgs tries to interact with request object directly, won't it be a problem? Also Integ tests are not Rest layer tests. And OpenSearch uses http smoke test pkg for rest integ tests. The current IT pkg doesn't support REST tests. Let me know if this makes sense..

Reply:
@imRishN I was wondering if you set the requestId here itself - https://github.com/opensearch-project/OpenSearch/blob/b71caaebf030a17050d00255bcefe43dba05e6ab/server/src/main/java/org/opensearch/rest/action/admin/cluster/RestDecommissionAction.java#L49 then it can be passed around as is.